mifosng-android/src/main/java/com/mifos/mifosxdroid/online/DashboardActivity.java

```diff
@@ -360,7 +361,19 @@ public boolean onPrepareOptionsMenu(Menu menu) {
 
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
+        final Intent intent = new Intent();
         switch (item.getItemId()) {
+            case R.id.menu_lock:
+                intent.setClass(this, PassCodeActivity.class);
+                startActivity(intent);
+                if (PrefManager.getPassCodeStatus()) {
+                    startActivity(new Intent(this, DashboardActivity.class));
+                } else {
+                    intent.setClass(this, PassCodeActivity.class);
+                    intent.putExtra(Constants.INTIAL_LOGIN, true);
+                    startActivity(intent);
+                }
```

Comment:
Don't you want to cover this sceanrio by test? Integration tests via espresso should cover this in the future